### PR TITLE
[Merged by Bors] - Warn instead of erroring when max_font_atlases is exceeded

### LINKED ISSUE #6642: `FontAtlasSet` is broken
## Bevy version

0.9.0

## What went wrong

Minimal repro:

```rust
const MAX_ATLASES: usize = 1;
const DYNAMIC_SIZE: bool = false;
const TEXT: &str = "";

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .insert_resource(TextSettings {
            max_font_atlases: NonZeroUsize::new(MAX_ATLASES).unwrap(),
            allow_dynamic_font_size: DYNAMIC_SIZE,
        })
        .add_startup_system(setup)
        .run();
}

fn setup(mut commands: Commands, asset_server: ResMut<AssetServer>) {
    commands.spawn(Camera2dBundle::default());

    commands.spawn(TextBundle::from_sections([TextSection::new(
        TEXT,
        TextStyle {
            font: asset_server.load("fonts/sharetech-mono-regular.ttf"),
            font_size: 24.0,
            color: Color::RED,
        },
    )]));
}
```
Consider the following configurations:

1.  
```rust
const MAX_ATLASES: usize = 1;
const DYNAMIC_SIZE: bool = false;
const TEXT: &str = "AB"; // any text with 2 or more distinct characters
```
the application panics with: 

> 'Fatal error when processing text: exceeded 1 available TextAltases for font. This can be caused by using an excessive number of font sizes or ui scaling...

even though it's using only one font size.

2.
```rust
const MAX_ATLASES: usize = 2;
const DYNAMIC_SIZE: bool = true;
const TEXT: &str = "AB"; // any text with `MAX_ATLASES or more` distinct characters
```
the application panics with:

> thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/bevy_ui/src/render/mod.rs:362:22

3.
```rust
const MAX_ATLASES: usize = 1;
const DYNAMIC_SIZE: bool = true;
const TEXT: &str = "A"; // any non-empty text
```
the application doesn't panic, but it enters an endless loop.

I gave it a quick glance, and it seems that these lines:
https://github.com/bevyengine/bevy/blob/6763b31479246afddecbe4e717e138d4e9653dca/crates/bevy_text/src/font_atlas_set.rs#L57-L70
https://github.com/bevyengine/bevy/blob/6763b31479246afddecbe4e717e138d4e9653dca/crates/bevy_text/src/font_atlas_set.rs#L85

shouldn't be executed unconditionally, but only if the font with the given size isn't already present in `FontAtlasSet` (1st and 2nd example).

Also, the extra `- 1` on this line:
https://github.com/bevyengine/bevy/blob/6763b31479246afddecbe4e717e138d4e9653dca/crates/bevy_text/src/font_atlas_set.rs#L65
seems to be the cause of the endless loop in the third example.

### PR DESCRIPTION
**Pull request by rparrett**
# Objective

Fixes #6642

In a way that doesn't create any breaking changes, as a possible way to fix the above in a patch release.

## Solution

Don't actually remove font atlases when `max_font_atlases` is exceeded. Add a warning instead.

Keep `TextError::ExceedMaxTextAtlases` and `TextSettings` as-is so we don't break anything.

This is a bit of a cop-out, but the problems revealed by #6642 seem very challenging to fix properly.

Maybe follow up later with something more like https://github.com/rparrett/bevy/commits/remove-max-font-atlases later, if this is the direction we want to go.

## Note

See previous attempt at a "simple fix" that only solved some of the issues: #6666

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_text/src/error.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ pub enum TextError {
     NoSuchFont,
     #[error("failed to add glyph to newly-created atlas {0:?}")]
     FailedToAddGlyph(GlyphId),
+    // TODO this is not used. Remove it.
```
Comment (Carter0) [inline crates/bevy_text/src/error.rs:10]: If it isn't used, then why not just remove it now? Is it too out of scope for the PR?

Comment (alice-i-cecile) [reply crates/bevy_text/src/error.rs:10]: It would introduce a breaking change, which means this can't be shipped in Bevy 0.9.

**Review by Weibye — APPROVED** [review]
I think this looks good for the short term. We should probably also change the UiScale example to use a set of fixed values instead of a smooth interpolation between changes, but that should be done separately. (And can come after 0.9.1)

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**